### PR TITLE
Do not try to guess when to allocate a TTY and keep it as default

### PR DESCRIPTION
**Pull request by ulyssessouza**
**What I did**
Do not try to guess when to allocate a TTY and keep it as default

**Related issue**
Resolves https://github.com/docker/compose/issues/8908

### CODE REVIEW

**Comment by ndeloof** [comment]
I can't find a better fix for this, but still sad we have to give up with tty detection, and wonder this will trigger more issues from users who expect compose to automatically adjust to environment :-/

File: cmd/compose/run.go [hunk 1]
```diff
@@ -144,7 +144,7 @@ func runCommand(p *projectOptions, backend api.Service) *cobra.Command {
 	flags.StringArrayVarP(&opts.environment, "env", "e", []string{}, "Set environment variables")
 	flags.StringArrayVarP(&opts.labels, "label", "l", []string{}, "Add or override a label")
 	flags.BoolVar(&opts.Remove, "rm", false, "Automatically remove the container when it exits")
-	flags.BoolVarP(&opts.noTty, "no-TTY", "T", notAtTTY(), "Disable pseudo-noTty allocation. By default docker compose run allocates a TTY")
+	flags.BoolVarP(&opts.noTty, "no-TTY", "T", false, "Disable pseudo-noTty allocation. By default docker compose run allocates a TTY")
```
Comment (thaJeztah) [inline cmd/compose/run.go:147]: Same logic is also used in `exec` - should it be removed there as well?

Comment (ulyssessouza) [reply cmd/compose/run.go:146]: I think so... That would be strange to have a different behaviour.

**Comment by ulyssessouza** [comment]
@ndeloof Yep. I also tried to save the automatic behaviour, but without success...
But at least the possible issues will have just a "-T" as "fix"

**Review by thaJeztah — APPROVED** [review]
SGTM

just to double-check; this brings it to the same behaviour / defaults as compose v1, correct?

**Review by ndeloof — APPROVED** [review]

**Comment by rfay** [comment]
Since panic is a result of this I don't think it's probably satisfactory, see https://stackoverflow.com/questions/70855915/fix-panic-provided-file-is-not-a-console-from-docker-compose-in-github-action

**Comment by ndeloof** [comment]
I don't like this either. I expect we get rid of this as we share more code with docker/cli, especially relying on the exact same `RunExec` function

**Comment by rfay** [comment]
Right... but not preventing a panic? that doesn't seem like the right thing...

**Comment by ndeloof** [comment]
We can burn some cycles trying to find a better way to manage this and avoid a `panic`, which is obviously the worst possible user experience to offer :'( 
but with https://github.com/docker/compose/pull/9198 and https://github.com/docker/cli/pull/3436 we are close to just drop all the custom `exec` code implemented by docker/compose to adopt the battle-tested `RunExec` code used by the docker CLI for years.

**Comment by rfay** [comment]
Sounds good.  I had actually planned long ago and half-implemented a switch to using docker's exec for `ddev exec` instead of using `docker-compose exec` but didn't actually pull the trigger, just worried about unexpected results.

**Comment by rostislav-simonik-plc** [comment]
Hello, i just tested with version 2.6.1 of docker compose and still having the issue. Once i remove exec with tee it starts working.

**Pull request merged**